### PR TITLE
[inductor] add conv_transpose2d unary fusion for cpu in inference mode

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90267
* #90266
* __->__ #90265
* #90264

An FX transformation is added to fuse ConvTranspose2d with eltwise OPs in torchinductor for CPU in inference mode, following the implementation in https://github.com/pytorch/pytorch/pull/87063.

The fusion OP is implemented in https://github.com/pytorch/pytorch/pull/90264 and will be treated as an extern kernel call in torchinductor.

The fusion of ConvTranspose2d with the below OPs is supported:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -3411,6 +3411,70 @@ def _prepare_convolution_fusion_create(
     return inputs, constant_args, kernel_layout, req_stride_order
 
 
+def _prepare_convolution_transpose_fusion_create(
+    cls,
```
Comment (XiaobingSuper) [inline torch/_inductor/ir.py:3415]: Could we combine this code with convolution?

Comment (chunyuan-w) [reply torch/_inductor/ir.py]: Updated the code to reuse the `_prepare_convolution_fusion_create` function for `ConvTranspose`.

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -1792,6 +1792,52 @@ def forward(self, x, y):
...
 
+    def test_conv_transpose2d_unary(self):
+        test_memory_format = [torch.contiguous_format, torch.channels_last]
+        options = itertools.product(
+            unary_list,
+            [True, False],
+            [1, 3],
+            [1, 2],
+            [1, 4],
+            [0],
```
Comment (jgong5) [inline test/inductor/test_torchinductor.py:1803]: Cover non-zero padding as well?

Comment (chunyuan-w) [reply test/inductor/test_torchinductor.py]: Changed padding from `[0]` to `[0, 1]` to covert the non-zero padding case.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Comment by chunyuan-w** [comment]
Re-opened in https://github.com/pytorch/pytorch/pull/91954

**Pull request closed without merging**